src/test/emulators/auth/setAccountInfo.spec.ts

```diff
@@ -415,6 +416,416 @@ describeAuthEmulator("accounts:update", ({ authApi, getClock }) => {
+      .set("Authorization", "Bearer owner")
+      .send({
+        localId,
+        mfa: {
+          enrollments: [mfaInfo],
+        },
+      })
+      .then((res) => expectStatusCode(200, res));
+
+    info = await getAccountInfoByIdToken(authApi(), idToken);
+    expect(info.mfaInfo).to.have.length(1);
+    const updated = info.mfaInfo?.pop();
+    expect(updated?.displayName).to.eq("New Display Name");
+    expect(updated?.phoneInfo).to.eq("+15555550101");
+    expect(updated?.mfaEnrollmentId).to.eq(mfaInfo.mfaEnrollmentId);
+  });
+
+  it("should allow changing the MFA enrollment ID", async () => {
+    let mfaInfo: any = {
+      displayName: "Cell Phone",
+      phoneInfo: TEST_PHONE_NUMBER,
+    };
+    const user = { email: "bob@example.com", password: "notasecret", mfaInfo: [mfaInfo] };
+    const { localId, idToken } = await registerMfaUser(authApi(), user);
+    let info = await getAccountInfoByIdToken(authApi(), idToken);
+    expect(info.mfaInfo).to.have.length(1);
+    mfaInfo = info.mfaInfo!.pop();
+    expect(mfaInfo?.mfaEnrollmentId).to.be.a("string").and.not.empty;
+
+    const newEnrollmentId = randomId(28);
+    await authApi()
+      .post("/identitytoolkit.googleapis.com/v1/accounts:update")
+      .set("Authorization", "Bearer owner")
+      .send({
+        localId,
+        mfa: {
+          enrollments: [{ ...mfaInfo, mfaEnrollmentId: newEnrollmentId }],
+        },
+      })
+      .then((res) => expectStatusCode(200, res));
+
+    info = await getAccountInfoByIdToken(authApi(), idToken);
+    expect(info.mfaInfo).to.have.length(1);
+    const updated = info.mfaInfo?.pop();
+    expect(updated?.displayName).to.eq(mfaInfo.displayName);
+    expect(updated?.phoneInfo).to.eq(mfaInfo.phoneInfo);
+    expect(updated?.mfaEnrollmentId).not.to.eq(mfaInfo.mfaEnrollmentId);
+    expect(updated?.mfaEnrollmentId).to.eq(newEnrollmentId);
+  });
+
+  it("should overwrite existing MFA info", async () => {
+    const mfaInfo: any = {
+      displayName: "Cell Phone",
+      phoneInfo: TEST_PHONE_NUMBER,
+    };
+    const user = {
+      email: "bob@example.com",
+      password: "notasecret",
+      mfaInfo: [mfaInfo, mfaInfo, mfaInfo],
+    };
+    const { localId, idToken } = await registerMfaUser(authApi(), user);
+    let info = await getAccountInfoByIdToken(authApi(), idToken);
+    expect(info.mfaInfo).to.have.length(3);
+
+    const newMfaInfo = {
+      displayName: "New New",
+      phoneInfo: "+12813308004",
+      mfaEnrollmentId: randomId(28),
+    };
+    await authApi()
+      .post("/identitytoolkit.googleapis.com/v1/accounts:update")
+      .set("Authorization", "Bearer owner")
+      .send({
+        localId,
+        mfa: {
+          enrollments: [newMfaInfo],
+        },
+      })
+      .then((res) => expectStatusCode(200, res));
+
+    info = await getAccountInfoByIdToken(authApi(), idToken);
+    expect(info.mfaInfo).to.have.length(1);
+    const updated = info.mfaInfo?.pop();
```

Comment:
Don't mutate the array here, or your for-loop below won't work as you expect.
```suggestion
    const updated = info.mfaInfo[0];
```